File: WebApplication1/Database/Interview.cs
using System.ComponentModel;
using System.Runtime.InteropServices.JavaScript;

namespace WebApplication1.Database;

public class Interview
{
    public Guid Id { get; set; }
    public virtual Vacancy Vacancy { get; set; }
    public virtual Student Student { get; set; }
    public virtual VacancyResponse VacancyResponse { get; set; }
    public DateTime DateTime { get; set; } 
    
    public InterviewResult? Result { get; set; }
}

public enum InterviewResult
{
    [Description("Прошел успешно")]
    Passed,
    [Description("Не прошел")]
    Failed,
    [Description("Отменено")]
    Canceled,
    [Description("Не явился")]
    DidNotCome,
}

File: Frontend/src/types/Interview.ts
import {Vacancy} from "./Vacancy.ts";
import {Student} from "./Student.ts";
import {VacancyResponse} from "./VacancyResponse.ts";

export type Interview = {
    id: string
    vacancy: Vacancy
    student: Student
    vacancyResponse: VacancyResponse
    dateTime: Date,
    result?: InterviewResult
}

export enum InterviewResult {
    Passed = 0,
    Failed = 1,
    Canceled = 2,
    DidNotCome = 3
}